go_modules/lib/dependabot/go_modules/update_checker/latest_version_finder.rb

```diff
@@ -94,8 +94,8 @@ def fetch_available_versions
               env = { "GOPRIVATE" => @goprivate }
 
               versions_json = SharedHelpers.run_shell_command(
-                "go list -m -versions -json #{dependency.name}",
-                fingerprint: "go list -m -versions -json <dependency_name>",
+                "go list -e -m -versions -json #{dependency.name}",
```

Comment:
Will this prevent `SharedHelpers::HelperSubprocessFailed` from being raised for the conditions handled in https://github.com/dependabot/dependabot-core/pull/8702/files#diff-9536ab82fee1fffd8e6fe35594bcddae3979d6de56792638b57d551856fff01eR109-R115?

If so, we'd need to convert that handling to use the form of error output or handle this particular error via exception instead?

Reply:
You're right, I think the short term solution is just to add it as another resolvability error instead.

Reply:
Actually now that I've looked at the error handling, I'm wondering why we're going to so much trouble? If we use `-e` then take any errors out of the JSON and raise them as DependencyFileNotResolvable that's much simpler. Then we don't have to keep updating this code every time a new error message pops up.

What do you think?

Reply:
Makes sense to me. Trying to identify errors based on the message isn't ideal.